pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -609,8 +609,15 @@ func (c *Cluster) findAvailableZoneIfStretched(mons []*monConfig) (string, error
 		zoneCount[m.Zone]++
 	}
 
+	var zones []cephv1.MonZoneSpec
+	if c.spec.IsStretchCluster() {
+		zones = c.spec.Mon.StretchCluster.Zones
+	} else {
+		zones = c.spec.Mon.Zones
+	}
+
 	// Find a zone in the stretch cluster that still needs an assignment
-	for _, zone := range c.spec.Mon.StretchCluster.Zones {
+	for _, zone := range zones {
 		count, ok := zoneCount[zone.Name]
 		if !ok {
 			// The zone isn't currently assigned to any mon, so return it
```

Comment:
This still looks like an issue (now see line 630 instead of 626). The stretch cluster chooses two mons per zone, but for non-stretch clusters, we only want one mon per zone. I believe we should update this condition
```go
if c.spec.Mon.Count == 5 && count == 1 && !zone.Arbiter {
```
to this:
```go
if c.spec.IsStretchCluster() && c.spec.Mon.Count == 5 && count == 1 && !zone.Arbiter {
```